src/Concordium/Client/Runner.hs

```diff
@@ -545,60 +545,62 @@ checkAndGetMemo memo pv = do
-              => Maybe FilePath
-              -> Maybe Types.BlockHash
-              -> Types.Event
-             -> ClientMonad m (Maybe CS.SchemaType)
-getSchemaType schemaFile blockHashM ev = do
+getContractInfoWithSchemas :: (MonadIO m, MonadFail m)
+              => Maybe FilePath -- ^ Path pointing to a schema file.
+              -> Maybe Types.BlockHash -- ^ Blockhash of the block to retrieve the contract info from.
+              -> Types.Event -- ^ The event for which the contract info will be retrieved.
+             -> ClientMonad m (Maybe CI.ContractInfo)
+getContractInfoWithSchemas schemaFile blockHashM ev = do
   -- Get contract address.
   let cAM = case ev of
               Types.ContractInitialized{..} -> Just ecAddress
               Types.Updated{..} -> Just euAddress
               Types.Interrupted{..} -> Just iAddress
               _ -> Nothing
-  -- Get schema.
-  schema <- case cAM of
+  -- Get module schema.
+  case cAM of
     Just ca -> do
       let blockHashTextM = Text.pack . show <$> blockHashM
       withBestBlockHash blockHashTextM $ \bh -> do
         contrInfo <- getContractInfo (NamedContractAddress ca []) bh
-        let cName = CI.ciName contrInfo
         let namedModRef = NamedModuleRef { nmrRef = CI.ciSourceModule contrInfo, nmrNames = [] }
-        getSchemaFromFileOrModule schemaFile namedModRef bh >>= \case
-          Just (CS.ModuleSchemaV3 m) -> case m Map.!? cName of
-            Just CS.ContractSchemaV3{..} -> return $ Just cs3EventSchema
-            Nothing -> return Nothing
-          _ -> return Nothing
+        schemaM <- getSchemaFromFileOrModule schemaFile namedModRef bh
+        case schemaM of
+          Nothing -> return $ Just contrInfo
+          Just schema -> CI.addSchemaData contrInfo schema
     _ -> return Nothing
-  return $ join schema
-
--- |Get the transaction events and their associated contract module V3 event schemas if
--- any.
--- Returns a map from blockhashes of blocks in which the transaction is present to
--- events of the transaction in that block. Each event figures in a pair with
--- an optional event schema of the V3 contract schema associated with in that block,
--- if any.
--- Optionally takes a path to a schema file to be parsed and returned. The schema from
--- the file will then take precedence over embedded schemas in the module and will thus
--- be included in its place for all events.
-getTxEventsAndSchemas :: (MonadIO m, MonadFail m)
-               => Maybe FilePath
-               -> TransactionStatusResult
-               -> ClientMonad m (Map.Map Types.BlockHash [(Types.Event, Maybe CS.SchemaType)])
-getTxEventsAndSchemas schemaFile status = do
+
+-- |Get `ContractInfo` for all events in all blocks in which a transaction is present.
+-- Returns a map from blockhashes of blocks in which the transaction is present to the
+-- events of the transaction in that block. Each event figures in a pair with an optional
```

Comment:
Figures?

Reply:
Sorry, this is Dan(ish-en)glish. I changed it.